src/realm_dart.cpp

```diff
@@ -108,19 +108,16 @@ void _callback(realm_userdata_t userdata, const realm_http_request_t request, vo
         std::vector<realm_http_header_t> headers_vector;
     } buf;
 
-    realm_http_request_t request_copy = request; // copy struct
-
     buf.url = request.url;
-    request_copy.url = buf.url.c_str();
-    buf.body = std::string(request.body, request.body_size);
-    request_copy.body = buf.body.data();
-
-    buf.headers_vector.reserve(request.num_headers);
```

Comment:
I would keep the reserve to ensure we don't need to regrow on a later push_back

Reply:
This is a premature optimization which was causing the crash.

Reply:
let's leave optimizations for after we have identified performance issues.

Reply:
No, the crash was due to the way the code assumed that `buf.headers_vector[i]` actually was already a `realm_http_header_t`. Lets keep the reserve.

Reply:
Calling reserve here brings tiny, unmeasurable and negligible performance difference which even Core is not doing and is not worth the discussion. I would like to merge this PR.

Reply:
It adds memory pressure, because vector has no way of knowing how much memory will be needed. But if you feel strongly for it, skip it..